pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/rest/InvalidPathResource.java

```diff
@@ -153,9 +167,39 @@ private String parseVersion(String path) {
     }
 
     /**
+     * Parses the version part from the path.
```

Comment:
These are private methods used only by wrongPath(). Having an enum and separate function seems to be an overkill for this. will do away with helper functions and use pathSegments directly to extract the version and endpoint.